enhancements/multi-arch/multiarch-manager-operator.md

```diff
@@ -0,0 +1,883 @@
+
+The following diagram reflects the proposed deployment model.
+
+```mermaid
+%%{init: {'securityLevel': 'loose', 'theme':'neutral', 'themeVariables': { 'fontSize':'14px', 'fontFamily':'Roboto Mono'}}}%%
+
+classDiagram
+    direction LR
+
+    class PodPlacementController {
+        <<Deployment (active-passive)>>
+        replicas=2
+    }
+
+    class Operator {
+        <<Deployment (active-passive)>>
+        replicas=2
+
+    }
+
+    class SchedulingGateMutatingWebhookConfiguration {
+        <<MutatingWebhookConfiguration>>
+        operations = [CREATE]
+        sideEffects = None
+        failurePolicy = Ignore
+    }
+
+    class PlacementPodConfig {
+        <<CR>>
+
+    }
+
+    class Pod {
+        schedulingGates: List
+        affinity: Object
+    }
+
+    class SchedulingGateMutatingWebhook {
+        <<Deployment (active-active)>>
+        replicas=3
+    }
+    SchedulingGateMutatingWebhookSvc --> SchedulingGateMutatingWebhook
+
+    class SchedulingGateMutatingWebhookSvc {
+        <<service>>
+    }
+
+    SchedulingGateMutatingWebhookConfiguration --> SchedulingGateMutatingWebhookSvc
+    Operator --> PodPlacementController : creates and reconciles
+    Operator --> SchedulingGateMutatingWebhook : creates and reconciles
+    Operator --> PlacementPodConfig : reconciles
+    PodPlacementController --> Pod : updates (patches nodeAffinity,<br/>removes schedulingGate)
+    SchedulingGateMutatingWebhook --> Pod : updates (adds schedulingGate)
+
+```
+### Open Questions
+- What is the performance impact when pods are gated and inspected over many
+  namespaces? This needs to be qualified. Mitigations have already been
+  discussed in the other sections, but we expect others to address at
+  implementation and performance evaluation time. Leader election time
+  should also be considered when pods are gated and the primary controller is
+  down. This would add seconds worth of delay to the scheduling. Is there
+  scope to minimize this?
+- Should a backoff retry mechanism with a maximum number of retries be
+  utilized in instances of image inspection failure (e.g., temporery registry
+  unreachability, network partitioning, etc.) instead of immediately removing the scheduling gate?
+- Do we need to be aware of any unique ways to access pull secrets for
+  disconnected clusters?
+- Is a cleanup mechanism needed to unblock pods that are scheduling gated upon
+  the operator's uninstallation or the CR's deletion?
+- Should the `SchedulingGateMutatingWebhook` and the `PodPlacementConfig` CR also consider a Pod Selector to filter
+  pods that should be gated and inspected? This would allow for a more fine-grained control of the pods.
+  The current proposal is to use a namespace selector to filter the namespaces for which the
+  operator should gate and inspect pods.
```

Comment:
i think there needs to be a way to opt out of this behavior, though that might be as simple as "provide your own node selectors" since this EP elsewhere seems to imply if the pod already has a nodeselector, this controller won't touch it, correct?

Reply:
yeah i think that's ok.  i'll resolve this thread.